x-pack/platform/plugins/shared/fleet/public/applications/fleet/sections/settings/components/download_source_flyout/ssl_form_section.tsx

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { EuiTextArea, EuiFormRow, EuiCallOut, EuiSpacer, EuiPanel, EuiTitle } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { i18n } from '@kbn/i18n';
+
+import { MultiRowInput } from '../multi_row_input';
+
+import { SecretFormRow } from '../edit_output_flyout/output_form_secret_form_row';
+
+import type { DownlaodSourceFormInputsType } from './use_download_source_flyout_form';
+
+interface Props {
+  inputs: DownlaodSourceFormInputsType;
+  useSecretsStorage: boolean;
+  isConvertedToSecret: boolean;
+  onToggleSecretAndClearValue: (secretEnabled: boolean) => void;
+}
+
+export const SSLFormSection: React.FunctionComponent<Props> = (props) => {
+  const { inputs, useSecretsStorage, isConvertedToSecret, onToggleSecretAndClearValue } = props;
+
+  return (
+    <>
+      <EuiPanel color="subdued" borderRadius="none" hasShadow={false}>
+        <EuiTitle size="s">
+          <h3 id="FleetEditOutputFlyoutKafkaAuthenticationTitle">
+            <FormattedMessage
+              id="xpack.fleet.settings.editDownloadSourcesFlyout.sslAuthenticationTitle"
+              defaultMessage="Authentication"
+            />
+          </h3>
+        </EuiTitle>
+        <EuiSpacer size="s" />
+        <EuiCallOut
+          title={i18n.translate('xpack.fleet.editDownloadSourcesFlyout.sslWarningCallout', {
+            defaultMessage:
+              'Invalid settings can prevent Elastic Agent from being able to upgrade. If this happens, you will need to provide valid credentials.',
+          })}
+          color="warning"
+          iconType="warning"
+        />
+        <EuiSpacer size="m" />
+        <MultiRowInput
+          placeholder={i18n.translate(
+            'xpack.fleet.settings.editDownloadSourcesFlyout.sslCertificateAuthoritiesInputPlaceholder',
+            {
+              defaultMessage: 'Specify certificate authority',
+            }
+          )}
+          label={i18n.translate(
+            'xpack.fleet.settings.editDownloadSourcesFlyout.sslCertificateAuthoritiesInputLabel',
+            {
+              defaultMessage: 'Server SSL certificate authorities (optional)',
```

Comment:
do we need `(optional)` here, all fields seems optional?

Reply:
I'm going to open a follow up PR to address some small UX changes across the whole feature. I think it can be addressed there.